tools/ImportTool.Shared/SplitBlobsCommand.cs

```diff
@@ -0,0 +1,151 @@
+            try
+            {
+                if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
+                {
+                    _logger.LogInformation("Start to split files in storage {0} with prefix {1}", account, prefix);
+
+                    CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+
+                    int count = await SplitBlob(
+                        cloudBlobClient,
+                        prefix,
+                        size,
+                        blockSize,
+                        maxConcurrentFile,
+                        maxSpliterPerBlob,
+                        maxUploaderPerBlob);
+
+                    _logger.LogInformation("Split {0} files in total", count);
+                }
+                else
+                {
+                    throw new StorageException(
+                        "An invalid connection string.");
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Failed to split due to {0}", ex.Message);
+            }
+        }
+
+        public static string GetConnectionStringFromAccountAndKey(string account, string key)
+        {
+            string storageConnectionString = "UseDevelopmentStorage=true;";
+            if (!(string.IsNullOrEmpty(account) || string.IsNullOrEmpty(key)))
+            {
+                storageConnectionString = $"DefaultEndpointsProtocol=https;AccountName={account};AccountKey={key}";
+            }
+
+            return storageConnectionString;
+        }
+
+        private static async Task<int> SplitBlob(
+            CloudBlobClient cloudBlobClient,
+            string prefix,
+            long sizeInBytes,
+            long blockSizeInBytes,
+            int maxConcurrentSplitFile,
+            int maxSpliterCountPerFile,
+            int maxUploaderCountPerSplitedFile)
+        {
+            int count = 0;
+            List<Task> runningTasks = new List<Task>();
+            BlobContinuationToken continuationToken = null;
+            do
+            {
+                var segments = await cloudBlobClient.ListBlobsSegmentedAsync(
+                      prefix: prefix,
+                      useFlatBlobListing: true,
+                      blobListingDetails: BlobListingDetails.Metadata,
+                      maxResults: 500,
+                      currentToken: continuationToken,
+                      options: null,
+                      operationContext: null);
+                continuationToken = segments.ContinuationToken;
+
+                foreach (var segment in segments.Results.Cast<CloudBlockBlob>())
+                {
+                    if (!segment.Name.StartsWith("splited/", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        _logger.LogDebug("Start to split files {0}", segment.Uri);
+
+                        if (runningTasks.Count >= maxConcurrentSplitFile)
```

Comment:
this will not cause error because .NET schedule the tasks and won't start too many threads at concurrent, but too much tasks may slowdown the program, here I add a limitation for this number, exceed input will downgrade to the maximum number.